internal/v2/fake_s2av2/fake_s2av2.go

```diff
@@ -0,0 +1,105 @@
+package fake_s2av2
+
+import (
+	"log"
+	"fmt"
+	_ "embed"
+	s2av2pb "github.com/google/s2a-go/internal/proto/v2/s2a_go_proto"
+	commonpb "github.com/google/s2a-go/internal/proto/v2/common_go_proto"
+)
+var (
+	//go:embed example_cert/client_cert.pem
+	clientCert []byte
+	//go:embed example_cert/server_cert.pem
+	serverCert []byte
+)
+
+type Server struct {
+	s2av2pb.UnimplementedS2AServiceServer
+}
+
+// SetUpSession receives SessionReq, performs request, and returns a
+// SessionResp, all on the server stream.
+func (s *Server) SetUpSession(stream s2av2pb.S2AService_SetUpSessionServer) error {
+	for {
+		req, err := stream.Recv()
+		if err != nil {
+			log.Fatalf("Fake S2A Service: failed to recieve SessionReq: %v", err)
+			return err
+		}
+		// Call one of the 4 possible RespOneof's
+		var s *s2av2pb.SessionResp
+		switch x := req.ReqOneof.(type) {
+		case *s2av2pb.SessionReq_GetTlsConfigurationReq:
+			s = GetTlsConfiguration(req.GetGetTlsConfigurationReq())
+		case *s2av2pb.SessionReq_OffloadPrivateKeyOperationReq:
+			// TODO(rmehta19): Implement fake.
+		case *s2av2pb.SessionReq_OffloadResumptionKeyOperationReq:
+			// TODO(rmehta19): Implement fake.
+		case *s2av2pb.SessionReq_ValidatePeerCertificateChainReq:
+			// TODO(rmehta19): Implement fake.
+		default:
+			return fmt.Errorf("SessionReq.ReqOneof has unexpected type %T", x)
+		}
+		err1 := stream.Send(s)
+		if err1 != nil {
+			log.Fatalf("Fake S2A Service: failed to send SessionResp: %v", err1)
+			return err1
```

Comment:
Thanks for pointing this out, I didn't know there was a difference between the two! I changed all the logging in SetUpSession to use Errorf, so that the error will be returned

Reply:
We'll use log.Printf as discussed offline